app/process_proposal.go

```diff
@@ -107,6 +107,13 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) abci.ResponsePr
 		}
 	}
 
+	if !data.Messages.IsSorted() {
```

Comment:
Note for reviewers, Github CI won't work but if you'd like to try this draft PR locally, you can pull the celestia-core change, build it locally, and then modify `celestia-app/go.mod`

```diff
replace (
	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v1.2.0-sdk-v0.46.0
	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
-	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.3.0-tm-v0.34.20
+	github.com/tendermint/tendermint => ../celestia-core
)
```

Reply:
why not use a specific commit? while it doesn't unblock this PR, it would allow the CI to run

Reply:
That's a good idea! Will use try this next time b/c CI should pass now that https://github.com/celestiaorg/celestia-app/pull/616 merged and this PR pulled in latest from `main`